### PR TITLE
Respect security configuration (#2815)

### PR DESCRIPTION
**Pull request by rbubley**

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @jcrist

**Comment by mrocklin** [comment]
Thanks for this fix @rbubley !  If it's ok with you I'd prefer to wait for @jcrist to take a look at this, but I suspect that he's off until next week due to the American holiday.

File: distributed/deploy/local.py [hunk 1]
```diff
@@ -123,11 +124,15 @@ def __init__(
...
         self.processes = processes
+        security = security or Security()
 
         if protocol is None:
             if host and "://" in host:
                 protocol = host.split("://")[0]
-            elif security:
+            elif (
+                security
+                and security.get_connection_args("scheduler")["require_encryption"]
```
Comment (jcrist) [inline distributed/deploy/local.py]: This should be:

```python
elif security and security.require_encryption:
```

**Comment by mrocklin** [comment]
This is in!  Thanks @rbubley for the fix and @jcrist for the review.

**Pull request merged**